nodeclient/http.go

```diff
@@ -135,6 +146,10 @@ func do(httpClient *http.Client, baseURL string, ctx context.Context, userInfo *
 		req.URL.User = userInfo
 	}
 
+	if requestHeaderHook != nil {
+		requestHeaderHook(req.Header)
+	}
+
```

Comment:
This should be moved to after the next block where the content-type was set, because you are modifying the headers after the hook again